exchange/liqui/liqui_endpoint.go

```
@@ -207,11 +223,17 @@ func (self *LiquiEndpoint) GetInfo(timepoint uint64) (exchange.Liqinfo, error) {
 	resp, err := client.Do(req)
 	if err == nil {
 		if resp.StatusCode == 200 {
-			defer resp.Body.Close()
-			resp_body, err := ioutil.ReadAll(resp.Body)
-			log.Printf("Liqui GetInfo response: %s", string(resp_body))
+			defer func() {
+				if err := resp.Body.Close(); err != nil {
+					log.Printf("Response body close error: %s", err.Error())
+				}
+			}()
+			respBody, err := ioutil.ReadAll(resp.Body)
```

the logic follow only handles err ==nil. There was no err!= nil handler yet.

---

If err != nil, we return err to client. This is normal behavior, you can check situation. Btw, this is inside liqui, old coding form, can be ignore anw.

---

it is common to handle the odd case first. Please handle err != nil case first.

---

fixed it

---

I still see the err = nil  case is handled first. We should return the error in odd case (err != nil) and handle the normal path after that.